### PR TITLE
Feature/paginate

### PR DESCRIPTION
**Pull request by jpaylaga**
Hello @mosufy 

This is my initial commit for review. Please note that code coverage is not yet complete and is still ongoing.

### CODE REVIEW

File: src/utils/paginate.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+/**
+ * Paginate Helper.
+ *
+ * @param sql
+ * @param sqlParams
+ * @param perPage
+ * @param page
+ * @returns {Promise<{pagination: {has_previous_page: boolean, next_page: number, has_next_page: *, num_items, num_pages: *, previous_page: number, current_page: number}, data: Promise<*>}>}
+ */
+export const paginate = async (sql, sqlParams, perPage, page) => {
```
Comment (mosufy) [inline src/utils/paginate.js:106]: Suggest to add 1 more optional parameter `totalCount`. This is for a use case where doing a `db.select(sql, sqlParams).length` might not be the best option (for cases with lots of columns and records). A dev might want to use a slightly more optimized query like `SELECT count(id) FROM 'table_name'` instead.

File: src/utils/__tests__/paginate.spec.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  it('should return an object containing limit and offset', () => {
+    const expected = {
+      limit: 10,
+      offset: 10,
+    };
+    const values = getLimitAndOffsetByPageAndContentPerPage(2, 10);
+    expect(values).toMatchObject(expected);
+  });
+
+  it('should throw an exception if page is undefined', () => {
```
[the remark this answers is not in the document]
Comment (jpaylaga) [reply src/utils/__tests__/paginate.spec.js]: Nice. I'll definitely use this one bro. Thank you :)

File: src/services/pagination/Paginator.js [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+  /**
+   * Constructor
+   *
+   * @param db
+   * @param sql
+   * @param sqlParams
+   * @param perPage
+   * @param currentPage
+   */
+  constructor(db, sql, sqlParams, perPage, currentPage = null) {
```
Comment (daison12006013) [inline src/services/pagination/Paginator.js:15]: @lordmangila @mosufy we can make the perPage by default as 10 and currentPage to 1, similar in laravel length aware.

Comment (jpaylaga) [reply src/services/pagination/Paginator.js]: Hello @daison12006013 I now have applied the suggestion :)

**Review by mosufy — APPROVED** [review]

**Comment by jpaylaga** [comment]
Thanks @mosufy :)

**Pull request merged**